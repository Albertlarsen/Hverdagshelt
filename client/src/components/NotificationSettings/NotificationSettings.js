// @flow

import React from 'react';
import jwt from 'jsonwebtoken';
import {Grid, Row, Col, ListGroup, ListGroupItem, Tabs, Tab} from 'react-bootstrap';
import {User} from "../../classTypes";
import Button from "react-bootstrap/es/Button";
import {MyIssuesNotificationSettingsForm} from "../MyIssuesNotificationSettingsForm/MyIssuesNotificationSettingsForm";
import {NotificationSettingsForm} from "../NotificationSettingsForm/NotificationSettingsForm";
import {NotificationSettingsMyCountiesForm} from "../NotificationSettingsMyCountiesForm/NotificationSettingsMyCountiesForm";

import css from './NotificationSettings.css';

interface State {
    decoded: Object,
    user: User,
    categoryChoises: number[],

}

export class NotificationSettings extends React.Component <State> {

    state = {
        decoded: jwt.verify(window.localStorage.getItem('userToken'), "shhhhhverysecret"),
        user: null,
        categoryChoises: []
    };



    render() {
        return(
            <Grid className="notifications">
            <Tabs defaultActiveKey={1} id="uncontrolled-tab-example">
                <Tab eventKey={1} title="Hjemkommune">
                    <NotificationSettingsForm />
                </Tab>
                <Tab eventKey={2} title="Mine kommuner">
                    <NotificationSettingsMyCountiesForm />
                </Tab>
                <Tab eventKey={3} title="Mine saker">
                    <MyIssuesNotificationSettingsForm />
                </Tab>
            </Tabs>
            </Grid>
        );
    }
}