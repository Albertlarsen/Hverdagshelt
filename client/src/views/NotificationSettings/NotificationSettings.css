.notifications{
    padding-top: 1em;
}