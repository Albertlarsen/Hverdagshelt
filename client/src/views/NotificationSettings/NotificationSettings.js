// @flow

import React from 'react';
import jwt from 'jsonwebtoken';
import {Grid, Row, Col, ListGroup, ListGroupItem, Tabs, Tab} from 'react-bootstrap';
import {User} from "../../classTypes";
import Button from "react-bootstrap/es/Button";
import {MyIssuesNotificationSettingsForm} from "../../components/MyIssuesNotificationSettingsForm/MyIssuesNotificationSettingsForm";
import {NotificationSettingsForm} from "../../components/NotificationSettingsForm/NotificationSettingsForm";
import {NotificationSettingsMyCountiesForm} from "../../components/NotificationSettingsMyCountiesForm/NotificationSettingsMyCountiesForm";

import css from './NotificationSettings.css';

interface State {
    decoded: Object,
    user: User,
    categoryChoises: number[],

}

export class NotificationSettings extends React.Component <State> {

    state = {
        categoryChoises: []
    };

    render() {
        return(
            <Grid className="notifications bottomFooter">
            <Tabs defaultActiveKey={1} id="uncontrolled-tab-example">
                <Tab eventKey={1} title="Mine kommuner">
                    <NotificationSettingsMyCountiesForm />
                </Tab>
                <Tab eventKey={2} title="Mine saker">
                    <MyIssuesNotificationSettingsForm />
                </Tab>
            </Tabs>
            </Grid>
        );
    }
}