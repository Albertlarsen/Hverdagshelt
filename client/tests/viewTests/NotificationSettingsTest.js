// @flow

import {shallow} from "enzyme/build";
import * as React from "react";
import {NotificationSettings} from "../../src/views/NotificationSettings/NotificationSettings";

describe('Test for NotificationSettings view', () => {
    const wrapper = shallow(<NotificationSettings />);

    it('renders correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });
});