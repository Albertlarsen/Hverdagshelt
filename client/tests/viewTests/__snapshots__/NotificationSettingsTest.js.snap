// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for NotificationSettings view renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NotificationSettings />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "bsClass": "container",
      "children": <Uncontrolled(Tabs)
        defaultActiveKey={1}
        id="uncontrolled-tab-example"
      >
        <Tab
          eventKey={1}
          title="Mine kommuner"
        >
          <NotificationSettingsMyCountiesForm />
        </Tab>
        <Tab
          eventKey={2}
          title="Mine saker"
        >
          <MyIssuesNotificationSettingsForm />
        </Tab>
      </Uncontrolled(Tabs)>,
      "className": "notifications bottomFooter",
      "componentClass": "div",
      "fluid": false,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Tab
            eventKey={1}
            title="Mine kommuner"
          >
            <NotificationSettingsMyCountiesForm />
          </Tab>,
          <Tab
            eventKey={2}
            title="Mine saker"
          >
            <MyIssuesNotificationSettingsForm />
          </Tab>,
        ],
        "defaultActiveKey": 1,
        "id": "uncontrolled-tab-example",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <NotificationSettingsMyCountiesForm />,
            "eventKey": 1,
            "title": "Mine kommuner",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <MyIssuesNotificationSettingsForm />,
            "eventKey": 2,
            "title": "Mine saker",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "bsClass": "container",
        "children": <Uncontrolled(Tabs)
          defaultActiveKey={1}
          id="uncontrolled-tab-example"
        >
          <Tab
            eventKey={1}
            title="Mine kommuner"
          >
            <NotificationSettingsMyCountiesForm />
          </Tab>
          <Tab
            eventKey={2}
            title="Mine saker"
          >
            <MyIssuesNotificationSettingsForm />
          </Tab>
        </Uncontrolled(Tabs)>,
        "className": "notifications bottomFooter",
        "componentClass": "div",
        "fluid": false,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Tab
              eventKey={1}
              title="Mine kommuner"
            >
              <NotificationSettingsMyCountiesForm />
            </Tab>,
            <Tab
              eventKey={2}
              title="Mine saker"
            >
              <MyIssuesNotificationSettingsForm />
            </Tab>,
          ],
          "defaultActiveKey": 1,
          "id": "uncontrolled-tab-example",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <NotificationSettingsMyCountiesForm />,
              "eventKey": 1,
              "title": "Mine kommuner",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <MyIssuesNotificationSettingsForm />,
              "eventKey": 2,
              "title": "Mine saker",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
